components/map/MapboxGlobe.tsx

```diff
@@ -36,26 +42,45 @@ const MapboxGlobe = () => {
       void router.events.off("beforeHistoryChange", handleRouteChange)
   }, [handleRouteChange, router, router.events])
 
-  const features: Array<Feature<Geometry, GeoJsonProperties>> = pipe(
-    entries,
-    A.filterMap((entry) =>
-      !entry.location?.geopoint
-        ? none
-        : some({
-            type: "Feature",
-            geometry: {
-              type: "Point",
-              coordinates: [
-                entry.location.geopoint.lng,
-                entry.location.geopoint.lat,
-              ],
-            },
-            properties: {
-              slug: entry.slug?.current ?? null,
-            },
-          })
+  const { patternNames } = useSelection()
+
+  const data = useMemo<FeatureCollection<Geometry, GeoJsonProperties>>(() => {
+    const features: Array<Feature<Geometry, GeoJsonProperties>> = pipe(
+      entries,
+      A.filter((entry) => {
+        if (patternNames.length === 0) return true
+        const patternNames2: string[] =
+          entry.patterns?.map((pattern) => pattern.name) ?? []
+
+        return patternNames.reduce((acc, v) => {
+          const foo = patternNames2.includes(v)
+          return acc && foo
+        }, true)
```

Comment:
nit: Could we use more meaningful variable names here for `foo` and `patternNames2`?